src/main/java/org/jabref/model/entry/Date.java

```diff
@@ -33,15 +33,16 @@ public Date(TemporalAccessor date) {
     public static Optional<Date> parse(String dateString) {
         List<String> formatStrings = Arrays.asList("uuuu-M-d", "uuuu-M", "d-M-uuuu", "M/uu", "M/uuuu", "MMMM d, uuuu", "MMMM, uuuu",
                 "d.M.uuuu", "uuuu.M.d", "uuuu");
-        for (String formatString : formatStrings) {
-            try {
-                TemporalAccessor parsedDate = DateTimeFormatter.ofPattern(formatString).parse(dateString);
-                return Optional.of(new Date(parsedDate));
-            } catch (DateTimeParseException ignored) {
-                // Ignored
+        if (dateString != null) {
```

Comment:
Either we do a Objects.requireNonNull and leave the checking to the caller, or we use this isBlank Method. Currently it will throw an NPE from the TemporalAccessor.parse method if null is passed.